### PR TITLE
Prompt when the destination readme has been edited and provide `--yes` flag for automation

### PR DESCRIPTION
**Pull request by infiniteluke**
**What**:
Add a warning if someone edited the README manually. Currently emdaer overwrites the file without warning. Also, new users often accidentally edit the README directly before realizing emdaer is installed.

I have it implemented as a warning that does not fail the process.
screen shot 2018-03-05 at 10 05 07 pm

It could be argued we should error here instead.

**Why**:
better ux

**How**:
execute git to check if the file being generated already exists and has a change.

**Checklist**:

- [x] Documentation
- [x] Tests
- [x] Ready to be merged 
- [x] Added myself to AUTHORS 
- [x] Change `--path` to emdaer taking space separated paths as per lint staged https://github.com/okonet/lint-staged/issues/138#issue-213736444.
- [x] Document path usage and lint-staged usage

### CODE REVIEW

**Comment by flipactual** [comment]
Yeah, we should error – emdaer is not able to do its thing, so, yeah, full on error

I'm not clear on what happens in the scenario where someone modifies `.emdaer/README.emdaer.md`, runs emdaer, then adds their changes and tries to commit – does this work assume that emdaer is only being run as a commit hook?

**Comment by infiniteluke** [comment]
> runs emdaer, then adds their changes and tries to commit

That's a good point @flipactual.. We may want to put this behind a flag. So the default would be that it does not check for changes. But, if you want emdaer to be `strict` you can enable strict mode with a flag? I was trying to avoid adding a flag because I like the opinionated nature of emdaer currently.

Still though, it seems like this feature could make incremental runs problematic. Is there a simple way to solve for this?

**Comment by flipactual** [comment]
The simplest method I can think of at the moment is to log with the error instructions like "hey, you have manual changes in this file, do a git checkout for that file and try again"

**Comment by infiniteluke** [comment]
Ok - I'm going to try something to that affect. But I also want to mention something about "moving changes to {NAME}.emdaer.md if you edited it manually"

**Comment by flipactual** [comment]
Yeah, that's good, just draw attention to the fact that you need to checkout the file before emdaer will be happy

**Comment by infiniteluke** [comment]
Another way to handle this would be with a prompt.

```
$ emdaer --checkDirtyDestination
  Are you sure you want to overwrite {NAME}.md. It has existing changes. y\n
$ y
  emdaer did the thing
```

a `--yes` flag could be used to force the overwrite for precommit hooks or other automation.

We could work out whether having the default be overwriting or prompting, but that'd be a decent solution.

Is that worth building before we the `plz git checkout that file` solution?

**Comment by flipactual** [comment]
Either way, I think this counts as a breaking change... if you want to go all the way with it, go for it

**Comment by infiniteluke** [comment]
If by default emdaer still overwrites, I don't think this is breaking. 

Should emdaer default to prompting or overwritting? I think UX tells me it should prompt by default since compiling readmes is new to most people. Maybe we can ask around a bit.

**Comment by flipactual** [comment]
I think it should default to prompting

And I think we should do a proper check to see if generated files have changed. We'll need to compare hashes instead of just checking git

**Comment by infiniteluke** [comment]
If the generated file hasn't changed should we simply not write the "new" one and skip logging?

**Comment by flipactual** [comment]
Sounds fine

**Comment by infiniteluke** [comment]
emdaer

**Comment by infiniteluke** [comment]
Open to wordsmithing plz

**Comment by infiniteluke** [comment]
I need to document this in the readme before we merge

**Comment by infiniteluke** [comment]
We shouldn't use git for tracking changes. We will store an md5 hash of the readme contents in a namespaced comment at the top of the file. Before emdaer runs, we will hash the content and compare it to the hash we found in the file. if there is a difference, we should prompt for overwriting.

After emdaer runs we will hash the generated contents, if that hash matches the prior hash, no changes occurred and we can skip writing the file.

**Comment by infiniteluke** [comment]
As discussed on zoom:

- partitioned prompts should be executed serially each awaiting the prior, while unprompted files are processed behind the scenes.
- logs should not be printed until prompts have been answered
- the above code is half refactored. ex: the `exitCode` isn't even returned at this point.
- hash implementation works but is currently in a broken state due to the stream wrangling I couldn't figure out.

**Comment by infiniteluke** [comment]
> partitioned prompts should be executed serially each awaiting the prior, while unprompted files are processed behind the scenes.

I did not complete this piece, but still think we can ship as is. I'm happy to open a ticket for this feature, as it will be a non breaking change.

I opened https://github.com/emdaer/emdaer/issues/97 for this

**Comment by infiniteluke** [comment]
# BREAKING CHANGE
@emdaer/cli now prompts the user if a manual change to the compiled README is detected. Use --yes to skip the prompt and automatically overwrite the README

Merging this PR should cut 3.0.0. But who knows 😂

File: packages/cli/src/cli.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
+/* @flow */
+
+const program = require('commander');
+// 💩
+const { promisify } = require('util');
+const glob = promisify(require('glob'));
+const { outputFile, readFile, readJson } = require('fs-extra');
+const inquirer = require('inquirer');
+const Promise = require('bluebird');
```
Comment (flipactual) [inline packages/cli/src/cli.js]: y tho?

Comment (infiniteluke) [reply packages/cli/src/cli.js]: `Promise.each` makes the sequential prompt feature super easy

Comment (flipactual) [reply packages/cli/src/cli.js]: Need this to happen... https://github.com/tc39/proposal-async-iteration

**Comment by infiniteluke** [comment]
_moved to main PR comment_

**Review by flipactual — APPROVED** [review]

**Pull request merged**